contracts/Identity.sol

```diff
@@ -1,15 +1,294 @@
-pragma solidity ^0.6.2;
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.9;
 
-import "./ERC734.sol";
-import "./IIdentity.sol";
+import "./interface/IIdentity.sol";
+import "./version/Version.sol";
+import "./storage/Storage.sol";
 
 /**
  * @dev Implementation of the `IERC734` "KeyHolder" and the `IERC735` "ClaimHolder" interfaces into a common Identity Contract.
+ * This implementation has a separate contract were it declares all storage, allowing for it to be used as an upgradable logic contract.
  */
-contract Identity is ERC734, IIdentity {
+contract Identity is Storage, IIdentity, Version {
+    bool private initialized = false;
+    bool private canInteract = true;
 
-    mapping (bytes32 => Claim) private claims;
-    mapping (uint256 => bytes32[]) private claimsByTopic;
+    constructor(address initialManagementKey, bool _isLibrary) public {
+        canInteract = !_isLibrary;
+
+        if (canInteract) {
+            __Identity_init(initialManagementKey);
+        } else {
+            initialized = true;
+        }
+    }
+
```

Comment:
description missing

Reply:
Done in https://github.com/onchain-id/solidity/pull/31/commits/4d65e15b725e8dadc1faf303153892af83567683